packages/core/src/utils/google-genai/index.ts

```diff
@@ -0,0 +1,346 @@
+            }
+            const result = originalMethod.apply(context, args);
+            // No response attributes for create (returns object of chat instance, not generated content)
+            return result;
+          } catch (error) {
+            captureException(error, {
+              mechanism: { handled: false, type: 'auto.ai.google_genai', data: { function: methodPath } },
+            });
+            throw error;
+          }
+        },
+      );
+    }
+
+    // Async/content-producing path
+    return startSpan(
+      {
+        name: `${operationName} ${model}`,
+        op: getSpanOperation(methodPath),
+        attributes: requestAttributes,
+      },
+      async (span: Span) => {
+        try {
+          if (finalOptions.recordInputs && args[0] && typeof args[0] === 'object') {
+            addPrivateRequestAttributes(span, args[0] as Record<string, unknown>);
+          }
+
+          const result = await originalMethod.apply(context, args);
+          addResponseAttributes(span, result, finalOptions.recordOutputs);
+          return result;
+        } catch (error) {
+          captureException(error, {
+            mechanism: { handled: false, type: 'auto.ai.google_genai', data: { function: methodPath } },
+          });
+          throw error;
+        }
+      },
+    );
+  };
+
+  return run;
+}
+
+/**
+ * Create a deep proxy for Google GenAI client instrumentation
+ * Recursively instruments methods and handles special cases like chats.create
+ */
+function createDeepProxy<T extends object>(target: T, currentPath = '', options?: GoogleGenAIOptions): T {
+  return new Proxy(target, {
+    get: (t, prop) => {
+      const value = Reflect.get(t, prop);
+      const methodPath = buildMethodPath(currentPath, String(prop));
+
+      if (typeof value === 'function' && shouldInstrument(methodPath)) {
+        // Special case: chats.create is synchronous but needs both instrumentation AND result proxying
+        if (methodPath === CHATS_CREATE_METHOD) {
+          const instrumentedMethod = instrumentMethod(value as (...args: unknown[]) => unknown, methodPath, t, options);
+          return function instrumentedAndProxiedCreate(...args: unknown[]): unknown {
+            const result = instrumentedMethod(...args);
+            // If the result is an object (like a chat instance), proxy it too
+            if (result && typeof result === 'object') {
+              return createDeepProxy(result as object, CHAT_PATH, options);
+            }
+            return result;
+          };
+        }
+
+        return instrumentMethod(value as (...args: unknown[]) => Promise<unknown>, methodPath, t, options);
+      }
+
+      if (typeof value === 'function') {
+        // Bind non-instrumented functions to preserve the original `this` context
+        return value.bind(t);
+      }
+
+      if (value && typeof value === 'object') {
+        return createDeepProxy(value as object, methodPath, options);
```

Comment:
```suggestion
        return createDeepProxy(value, methodPath, options);
```

is this typecast necessary? this should be narrowed already here...? 😅

Reply:
Nope, true!